collector/textfile.go

```diff
@@ -105,6 +106,14 @@ func convertMetricFamily(metricFamily *dto.MetricFamily, ch chan<- prometheus.Me
 			}
 		}
 
+		h := hash(metricFamily, metric)
+		if seenIn, ok := seen[h]; ok {
+			repr := friendlyString(*metricFamily.Name, names, values)
+			log.Warnf("Metric %s was read from %s, but has already been collected from file %s, skipping", repr, path, seenIn)
+			continue
```

Comment:
This is not a good idea, you're now randomly dropping metrics depending on the iteration order. We've had issues with this type of approach in the past.

Reply:
Yeah, it's not great, I agree. However, failing the scrape completely isn't that good either. Do you see some middle road or better approach?
Also, I just realized I've forgotten to set the error flag when this happens, so now it is not alertable. That needs to be fixed.

Reply:
Should we be failing the entire scrape, though? Skipping the textfile stuff completely seems okay (although I'll note that the metrics aren't _randomly_ dropped, iteration order is deterministic here, so which metrics are dropped is consistent across scrapes given consistent input), but any individual breakage from any collector resulting in `up=0` seems a bit harsh?

Reply:
Right, that would be my preference. It will require a bit of gymnastics though, since `promhttp` bails the entire scrape if we allow the data to get there.
I guess I'll do some sort of buffer to be able to detect failure before passing the metrics through the channel. Would this be something that should be done in node_exporter as well?

Reply:
That'd be a question for @SuperQ, I've heard no plans in that direction.